test/integration/kubernetes_deploy_test.rb

```diff
@@ -635,6 +635,19 @@ def test_bad_container_on_daemon_sets_fails
     ], in_order: true)
   end
 
+  def test_resource_quotas_are_deployed_first
+    forced_timeout = 10
+    KubernetesDeploy::Deployment.any_instance.stubs(:timeout).returns(forced_timeout)
+    result = deploy_fixtures("resource-quota")
+    assert_deploy_failure(result)
+    assert_logs_match_all([
+      "Predeploying priority resources",
+      "Deploying ResourceQuota/resource-quotas (timeout: 30s)",
+      "Deploying all resources",
+      "Deployment/web deployment timed out"
+    ], in_order: true)
+  end
```

Comment:
Let's also fetch the quota and assert something about its spec and/or status since we don't have a separate test for that.